src/Interpreters/convertFieldToType.cpp

```diff
@@ -660,7 +660,15 @@ static bool decimalEqualsFloat(Field field, Float64 float_value)
 
 std::optional<Field> convertFieldToTypeStrict(const Field & from_value, const IDataType & from_type, const IDataType & to_type, const FormatSettings & format_settings)
 {
-    Field result_value = convertFieldToType(from_value, to_type, &from_type, format_settings);
+    Field result_value;
+    try
+    {
+        result_value = convertFieldToType(from_value, to_type, &from_type, format_settings);
+    }
+    catch (...)
+    {
+        return {};
+    }
```

Comment:
I think catching any exception here is a bit excessive because such errors still occur in `system.errors`, which can sometimes be confusing. 
More importantly, we can catch logical errors, std exceptions, or memory limit errors. One option to address the second issue is to check the error code for `UNKNOWN_ELEMENT_OF_ENUM`.

However, a better solution would be to explicitly ensure that when converting a string to an enum, the string is part of the enum values. Since we already have an explicit check for decimal values here, it makes sense to include a similar check for enums.

Reply:
Done